scripts/setup_truefoundry_cluster.sh

```diff
@@ -87,23 +87,48 @@ function make_request() {
     return 0
 }
 
+
+function is_cluster_provisioned() {
+    log_info "is_cluster_provisioned: Checking if cluster exists and is provisioned..."
+
+    # Make GET request to check cluster status
+    local response=$(make_request "GET" "${CONTROL_PLANE_URL}/api/svc/v1/cluster/${CLUSTER_NAME}" "" "200") || {
+        log_info "is_cluster_provisioned: Got error while querying for cluster ${CLUSTER_NAME}"
+        echo "false"
+        return 0
```

Comment:
we should do a non-zero exit here

Reply:
I did try that, the TF external module doesn't like any non-exit return code, It would fail the whole TF apply altogether with that.

Reply:
I don't think it does while I was testing - although the script should fail right, shouldn't it ?